### PR TITLE
Tigera Calico Intergration

### PR DESCRIPTION
**Pull request by troy-ameigh**
*Description of changes: Intergrated Tigera Calico Network Policy (Community)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by troy-ameigh** [comment]
Made an update to fix issue checker found

File: .gitmodules [hunk 1]
```diff
@@ -42,3 +42,7 @@
 	path = submodules/quickstart-eks-hashicorp-consul
 	url = git@github.com:aws-quickstart/quickstart-eks-hashicorp-consul.git
 	branch = main
+
+[submodule "submodules/quickstart-eks-tigera-calico"]
+	path = submodules/quickstart-eks-tigera-calico
+	url = git@github.com:aws-quickstart/quickstart-eks-tigera-calico.git
```
Comment (gargana) [inline .gitmodules:48]: Where is the branch that this will map too?

File: .taskcat.yml [hunk 2]
```diff
@@ -87,6 +87,8 @@ tests:
       EKSPublicAccessEndpoint: Enabled
       NodeGroupType: 'Unmanaged'
       ALBIngressController: Disabled
+      PerAccountSharedResources: "Yes"
```
Comment (gargana) [inline .taskcat.yml:90]: This isn't correct for the CI can these be removed?

File: templates/amazon-eks.template.yaml [hunk 3]
```diff
@@ -723,6 +740,25 @@ Resources:
...
+      DependsOn:
+        - !Ref ControlPlaneSecurityGroupIngress
+        - !Ref NodeGroupStack
+        - !If [ EnableFargate, !Ref FargateNamespaces, !Ref 'AWS::NoValue' ]
+    Properties:
+      TemplateURL: !Sub
+        - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}submodules/quickstart-eks-tigera-calico/templates/eks-calico-entrypoint.template.yaml'
+        - S3Region: !If [ UsingDefaultBucket, !Ref 'AWS::Region', !Ref QSS3BucketRegion ]
+          S3Bucket: !If [ UsingDefaultBucket, !Sub '${QSS3BucketName}-${AWS::Region}', !Ref QSS3BucketName ]
+      Parameters:
```
Comment (gargana) [inline templates/amazon-eks.template.yaml:746]: Are the other Calico params not required here?

**Review by gargana — CHANGES_REQUESTED** [review]
Hi Troy

Some small cosmetic things and something in .taskcat.yml to address.

Otherwise looking good.

Andrew

File: templates/amazon-eks-entrypoint-existing-vpc.template.yaml [hunk 4]
```diff
@@ -92,6 +92,10 @@ Metadata:
         Parameters:
           - NewRelicIntegration
           - NewRelicLicenseKey
+      - Label:
+          default: Calico Policy [APN security partner]
+        Parameters:
```
[the remark this answers is not in the document]
Comment (troy-ameigh) [reply templates/amazon-eks-entrypoint-existing-vpc.template.yaml:97]: CalicoIntegration is the only needed parameter

[the remark this answers is not in the document]
Comment (troy-ameigh) [reply templates/amazon-eks-entrypoint-existing-vpc.template.yaml:97]: Commit 7a0292a  should have cleaned this up

File: templates/amazon-eks.template.yaml (hunk 3 above)
Comment (troy-ameigh) [reply templates/amazon-eks.template.yaml:746]: No you only need the enablement param

**Review by gargana — APPROVED** [review]

**Pull request merged**